chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -927,7 +934,12 @@ impl PeerManagerActor {
                 remote_addr,
                 peer_info,
                 connecting_status,
-                FramedWrite::new(write, Codec::default(), Codec::default(), ctx),
+                FramedWrite::new(
+                    write,
+                    Codec::new(write_buf_size.clone()),
+                    Codec::new(write_buf_size),
```

Comment:
add a comment explaining why this won't cause a race condition when setting the gauge.

Reply:
I would say the comment is not needed here -- this compiles, the gauge is a `: Sync + Clone` object, it is thread safe. 

The WTF why we need to codecs here is somewhat inherent in the current architecture, there's indeed a fine interplay between codec as encoder and codec as encoder callback, that is explained in the "this is hack" comment on the field.

Reply:
what I mean is to document why 2 instances setting the same gauge won't cause a flickering.